### ISSUE TITLE
Slackbot + Docker + ngrok results in Dispatch_failed

### ISSUE THREAD

**Issue by sharathsridhar** [osteel/docker-tutorial#10]
Hi @osteel, I wanted some inputs from you regarding Docker + ngrok as I am stuck and don't know how to proceed further. 

Tl;Dr Docker running in one terminal, fails to listen/recognize ngrok's url running on another terminal. How to make Docker listen to ngrok?

Running the slackbot (built in python + flask) within PyCharm works fine, as in commands are executed and payloads/modals are being sent received. And so does running the command `gunicorn --bind 0.0.0.0:5000 wsgi:flaskApp -w 4` within PyCharm's termnal (Please note, ngrok is running on mac's terminal via `./ngrok http 5000`). The url in "Interactivity & Shortcuts" AND "Slash Commands" both point to https://...ngrok.io/slack/events. Again, the app, the commands, ngrok, everything works fine.

But when I move the slack app to within Docker and start Docker up with `docker-compose up --build` which in turns runs `ENTRYPOINT ["gunicorn", "--bind", "0.0.0.0:5000", "wsgi:flaskApp", "--workers", "4"]`, the slackbot fails with `dispatch_failed` when a command is executed and ngrok terminal shows `502 Bad Gateway`. The docker command/image, itself runs fine without any errors, but for some reason it isn't able to recognize the url exposed by ngrok. Any ideas what's wrong?

Dockerfile

```FROM python:3.7
ENV PYTHONDONTWRITEBYTECODE 1
ENV PYTHONUNBUFFERED 1

WORKDIR /app

COPY requirements.txt ./

COPY Main.py wsgi.py EmailSender.py AppConfig.py CompareCopyFgData.py SlackPayload.py Queries.py ./

RUN pip install --upgrade pip && pip install --no-cache-dir -r requirements.txt

ENV FLASK_APP=Main.py
ENV FLASK_ENV=production

ENTRYPOINT ["gunicorn", "--bind", "0.0.0.0:5000", "wsgi:flaskApp", "--workers", "4"]
```

docker-compose.yml
```
version: "3.9"
services:
  slackbot:
    build: 
      context: App/
      dockerfile: Dockerfile
    env_file:
      - ./App/.env
```
#### Steps to reproduce:

1. Start ngrok in a terminal using `ngrok http 5000`
2. Set up docker-compose.yml and Dockerfile with the appropriate files and run commands
3. In a new terminal, point to the folder containing docker-compose.yml image and run command `docker-compose up --build`
4. If no errors, go to slack and run a slash command and wait to see logs in Docker Terminal window

### Expected result:

Docker to accept incoming slack commands and print logs, send modal(s) to slack and receive user inputs etc

### Actual result:

Nothing happens in docker. The 4 workers are started and keep waiting/listening for a payload/data to come in, even though ngrok is running.

<img width="739" alt="Screen Shot 2022-05-01 at 2 17 27 PM" src="https://user-images.githubusercontent.com/13225331/166166559-a2fe0b3e-086c-465b-ac08-3acdeba5be45.png">
<img width="848" alt="Screen Shot 2022-05-01 at 2 17 45 PM" src="https://user-images.githubusercontent.com/13225331/166166560-0ba4219f-3b43-4d18-ae20-b3dc4a191430.png">

**Comment by osteel** [comment]
Hi, closing this as it looks like you found a solution [here](https://github.com/slackapi/bolt-python/issues/642) and to be honest this is rather out of scope with regard to my tutorial series 🙂

**Comment by sharathsridhar** [comment]
I did, Thank you for looking into it @osteel!
